test/unit/job_invocation_composer_test.rb

```diff
@@ -925,12 +925,14 @@ def provider_inputs
 
     context 'with template in setting present' do
       before do
-        FactoryBot.create(
-          :setting,
-          :name => 'remote_execution_form_job_template',
-          :category => 'Setting::RemoteExecution',
-          :value => setting_template.name
-        )
+        Foreman.settings._add('remote_execution_form_job_template',
+          context: :test,
+          type: :string,
+          category: 'Setting::RemoteExecution',
+          full_name: '',
+          description: '',
+          default: '')
```

Comment:
this should not be needed, that should be already there, only settings that are added only in tests need to use this.
Simple `Setting[<name>] = value` should do the job for existing settings.